pandas/tests/frame/test_constructors.py

```diff
@@ -195,6 +195,14 @@ def test_constructor_overflow_int64(self):
         df_crawls = DataFrame(data)
         assert df_crawls['uid'].dtype == np.uint64
 
+    def test_constructor_overflow_uint64(self):
+        # see gh-18584
+        values = np.array([2**64], dtype=object)
+        result = DataFrame(values)
```

Comment:
This test already passes on master, it was when you didn't already have an object array that it failed:

```
In [6]: pd.DataFrame(np.array([2**64], dtype=object))
Out[6]: 
                      0
0  18446744073709551616

In [7]: pd.DataFrame([2**64])
...
OverflowError: Python int too large to convert to C unsigned long
```

(but keep also this one)

Reply:
> Can you also check the case of large negative integers ?

We do now. 😄 

Added new tests as well.